addrs/provider.go

```diff
@@ -47,3 +64,79 @@ func (pt Provider) LegacyString() string {
 	}
 	return pt.Type
 }
+
+// ParseProviderSourceString parses the source attribute and returns a provider.
+// This is intended primary to parse the FQN-like strings returned by terraform-config-inspect.
+func ParseProviderSourceString(str string) (Provider, tfdiags.Diagnostics) {
+	var ret Provider
+	var diags tfdiags.Diagnostics
+
+	parts := strings.Split(str, "/")
```

Comment:
A comment above this could be nice to clarify what "parts" are about to future readers